sim/battle-actions.ts

```diff
@@ -0,0 +1,1620 @@
+import {Dex, toID} from './dex';
+
+const CHOOSABLE_TARGETS = new Set(['normal', 'any', 'adjacentAlly', 'adjacentAllyOrSelf', 'adjacentFoe']);
+
+const MAX_MOVES: {readonly [k: string]: string} = {
+	Flying: 'Max Airstream',
+	Dark: 'Max Darkness',
+	Fire: 'Max Flare',
+	Bug: 'Max Flutterby',
+	Water: 'Max Geyser',
+	Status: 'Max Guard',
+	Ice: 'Max Hailstorm',
+	Fighting: 'Max Knuckle',
+	Electric: 'Max Lightning',
+	Psychic: 'Max Mindstorm',
+	Poison: 'Max Ooze',
+	Grass: 'Max Overgrowth',
+	Ghost: 'Max Phantasm',
+	Ground: 'Max Quake',
+	Rock: 'Max Rockfall',
+	Fairy: 'Max Starfall',
+	Steel: 'Max Steelspike',
+	Normal: 'Max Strike',
+	Dragon: 'Max Wyrmwind',
+};
+
+const Z_MOVES: {readonly [k: string]: string} = {
+	Poison: "Acid Downpour",
+	Fighting: "All-Out Pummeling",
+	Dark: "Black Hole Eclipse",
+	Grass: "Bloom Doom",
+	Normal: "Breakneck Blitz",
+	Rock: "Continental Crush",
+	Steel: "Corkscrew Crash",
+	Dragon: "Devastating Drake",
+	Electric: "Gigavolt Havoc",
+	Water: "Hydro Vortex",
+	Fire: "Inferno Overdrive",
+	Ghost: "Never-Ending Nightmare",
+	Bug: "Savage Spin-Out",
+	Psychic: "Shattered Psyche",
+	Ice: "Subzero Slammer",
+	Flying: "Supersonic Skystrike",
+	Ground: "Tectonic Rage",
+	Fairy: "Twinkle Tackle",
+};
+
+export class BattleActions {
+	battle: Battle;
+	dex: ModdedDex;
+	constructor(battle: Battle) {
+		this.battle = battle;
+		this.dex = battle.dex;
+		if (this.dex.data.Scripts.actions) Object.assign(this, this.dex.data.Scripts.actions);
+	}
```

Comment:
(For OMs)
```suggestion
	constructor(battle: Battle) {
		this.battle = battle;
		this.dex = battle.dex;
		if (this.dex.data.Scripts.actions) Object.assign(this, this.dex.data.Scripts.actions);
		if (battle.format.actions) Object.assign(this, battle.format.actions);
	}
```
(Unfortunately you can't use the pattern that other classes use because all gens have battle actions.)